### DISCUSSION TITLE
Feature Request: python client relational API functionality to rename PyRelation columns

### DISCUSSION THREAD

**Discussion by scheah92** [duckdb/duckdb#13400] [Ideas]
Both `pandas` and `pyspark` offer direct functionality to rename specific columns in their dataframes. In fact, `duckdb`'s own `pyspark` implementation [includes the `withColumnRenamed` functionality](https://github.com/duckdb/duckdb/blob/6660e89bb90ab799dadbf424cac2a7a02eb09179/tools/pythonpkg/duckdb/experimental/spark/sql/dataframe.py#L84) from base `pyspark`. Currently, to do this with `duckdb` you would need to do this
```python
rel = rel.select("""
   col1,
   col2,
   col3 as col3_aliased,
   col4,
   col5 as col5_aliased,
   ...
   """
)
```
This functionality is useful when working with relations with many columns and you want to do some ad-hoc renaming. Many data scientists coming from `pandas` or `pyspark` backgrounds would be used to being able to do this. Being able to do targeted column renaming via the relational API would make adoption easier by providing familiar functionality.

An example usage would be something like below, allowing for either single column renaming or multiple column renaming via a dict mapping
```python

# Renaming a single column
rel = rel.withColumnRenamed(targetCol = "col1", renamedCol = "col1_alias")

# Renaming multiple columns
rel = rel.withColumnsRenamed(
    {
        "col1": "col1_aliased", 
        "col3": "col3_aliased"
    }
)
```

**Comment by fabienval** [comment]
I would love this feature!

I cannot seem to rename the column generated by the "first" aggregation...
I have the following table and would like to rename "first"(scan_id) to just scan_id but to no avail...
loc | pit | "first"(scan_id)
-- | -- | --
ng7 | 5 | 72
ng7 | 1 | 66
ng7 | 4 | 70
ng7 | 7 | 75
